### ISSUE TITLE
Efficiency for Git in multiple versions

### ISSUE THREAD

**Issue by vanrein** [arpa2/mkhere#22]
Git can efficiently handle multiple repositories on one disk.  Objects are silently shared and therefore save storage, especially when many versions, branches and so on are used.

Doing this can be a matter of a shared repository, perhaps a bare repository in `/dl/PKG-VERSION[-VARIANT]`, and sharing it with the various checkouts in `/src/PKG-VERSION[-VARIANT]`.

It may be interesting to start working on generic support for things like CMake and Git, in `/mkhere/lib/git` and `/mkhere/lib/cmake`.

**Comment by vanrein** [comment]
The directory structure is also a bit strange,

```
DIR_SRC="${ALTCACHE}/src/${TARGET}-${VERSION}"
DIR_GIT="${DIR_SRC}/${TARGET}-${VERSION}.git"
```

We may simplify by just targeting Git at `$DIR_SRC` which it will happily do if it finds an empty directory.  Given local replication, that is a cheap (and fast) operation.

**Comment by vanrein** [comment]
Generic support moved to #24, "Library support for general patterns".

**Comment by vanrein** [comment]
Closed with e1c1143682e5cc35dc500fb9f3d456613afdb541 in branch `master`.
